### ISSUE TITLE
Onboard the Codewind operator to OperatorHub

### ISSUE THREAD

**Issue by johnmcollier** [eclipse/codewind#2694]
[OperatorHub](https://operatorhub.io/) provides a marketplace for Kubernetes operators, and is also included by default on OCP 4. Having Codewind onboarded to OperatorHub would provide a great way for users to discover Codewind, as well as provide an easier way to install and configure Codewind instances on Kubernetes

[There's a well defined process](https://operatorhub.io/contribute) for contributing operators to OperatorHub and as the Codewind operator matures, we should see what it would take to contribute Codewind to it.

**Comment by markcor11** [comment]
@johnmcollier I agree that would be cool. We have talked about doing that and had mentioned it in the initial design story. There are some enhancements we need to make to accommodate their process first though.

**Comment by johnmcollier** [comment]
@markcor11 Good point - curious, what enhancements would be needed to the operator?

**Comment by markcor11** [comment]
@johnmcollier some examples of what need to be part of this story:

1. add additional CR status to the LIST output to pass OLM and scorecard tests
2. gracefully handle upgrades of both operator and Codewind resources
3. make namespace watches optional
4. generate a single pair of CRDs using our lowest supported Kube API level (so that we don't have to catalogue multiple versions)
5. add additional CR input validation (where we can't use kubebuilder annotations directly because of our need to support older Kube api levels)
6. optimise the reconcile loop and dequeue failures where appropriate
